gitpod-io/gitpod
# [PAYG - dashboard]: Implement Personalised content selection for users based on profile

## Description


This PR enhances the `PersonalizedContent` component to provide truly personalized content for users based on their profile information and exploration reasons. The changes include:

- Implemented a new content selection algorithm that considers:
    - User's exploration reasons (prioritizing "working-professional" over "personal")
    - Signup goals
    - Job role

- Added logic to show different content sets:
    - First week: Personalized content based on user profile
    - After the first week: Random content from the entire list, avoiding repetition

- Improved localStorage usage:
    - Storing the last shown content to prevent repetition
    - Tracking time to determine content refresh

## Related Issue(s)

Fixes OPM-669

## How to test


- Open the preview environment and fill out the Onboarding form.
- See data based on your selections.
- After 1 iteration, you can re-update the selections by going on `/?onboarding=force`. But, remember, that existing service doesn't assume Goals and a unique thing, so your new selections would get added to the previous data, but if you "uncheck" (which is rare and next to impossible for ordinary users - 99%, because force onboarding endpoint is hidden) anything from form, user profile data.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - siddhant-o3bc04bba73</li>
	<li><b>🔗 URL</b> - <a href="https://siddhant-o3bc04bba73.preview.gitpod-dev.com/workspaces" target="_blank">siddhant-o3bc04bba73.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - siddhant-opm-669-make-personalised-for-you-actually-personalised-for-users-gha.27428</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-siddhant-o3bc04bba73%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
